util/tracing/detect/threadsafe.go

```diff
@@ -7,6 +7,27 @@ import (
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 )
 
+// Duplicated from buildkit/client.
+// Part of the external interface of our tracer.
+// Needed to ensure that we maintain this interface.
+type tracerDelegate interface {
+	sdktrace.SpanExporter
+
+	SetSpanExporter(ctx context.Context, exp sdktrace.SpanExporter) error
+}
+
+func makeThreadSafe(exp sdktrace.SpanExporter) sdktrace.SpanExporter {
+	d, isDelegate := exp.(tracerDelegate)
```

Comment:
Maybe this should detect the `*jaeger.Exporter` instead and only make that thread-safe. Delegated exporter is already thread-safe.

Reply:
Or actually just wrap it in `jaegerExporter`, not in the `detect` pkg.

Reply:
ya, maybe the real solution is to complain upstream about the jaeger exporter not being thread-safe. in any case, i moved this code into the `jaeger` package.